magento/magento2
# #7241: Always add empty option for prefix and/or suffix if optional

### Description
This solves part of #7241. If the prefix and/or suffix field is set to "optional" in the configuration, we automatically add a blank first option if it isn't entered in the "Dropdown Options" configuration field. 

### Fixed Issues (if relevant)
1. magento/magento2#7241: No option to start with blank option for prefix and suffix in checkout. (partly fixed)

### Manual testing scenarios

1. Go to Admin panel > Stores > Configuration > Customers > Customer Configuration
2. Change "Show Prefix" and/or "Show Suffix" to "Optional"
3. Populate "Prefix Dropdown Options" with : "Ms;Miss;Mrs;Mr;Dr;Atty;Hon;Prof;Pres;VP;MA"
4. Populate "Suffix Dropdown Options" with : "Jr;Sr;Ph.D.;M.D.;R.N.;Ret."
5. Click Save.
6. Switch to the front-end, and add something to your cart.
7. Go to the checkout page or customer registration page
8. You'll notice that there is an empty first option for the Prefix and Suffix field.


### Contribution checklist
